app/src/main/java/com/github/lookupgroup27/lookup/ui/quiz/Quiz.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.lookupgroup27.lookup.ui.quiz
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.blur
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.lookupgroup27.lookup.R
+import com.github.lookupgroup27.lookup.ui.navigation.NavigationActions
+import com.github.lookupgroup27.lookup.ui.navigation.Screen
+
+@Composable
+fun QuizScreen(navigationActions: NavigationActions) {
+  Box(
+      modifier = Modifier.fillMaxSize().testTag("quiz_screen"),
+      contentAlignment = Alignment.Center,
+  ) {
+
+    // Background Image
+    Image(
+        painter = painterResource(id = R.drawable.landing_screen_bckgrnd),
+        contentDescription = "Background",
+        contentScale = ContentScale.Crop,
+        modifier = Modifier.fillMaxSize().blur(10.dp).testTag("quiz_background"))
+
+    IconButton(
+        onClick = { navigationActions.goBack() },
+        modifier =
+            Modifier.padding(16.dp).align(Alignment.TopStart).testTag("go_back_button_quiz")) {
+          Icon(
+              imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+              contentDescription = "Back",
+              tint = Color.White)
+        }
+
+    // Overlay for content
+    Column(
+        modifier = Modifier.align(Alignment.Center).padding(horizontal = 32.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(32.dp)) {
+          // Title Text
+          Text(
+              text = "Take a Quiz",
+              color = Color.White,
+              style =
+                  MaterialTheme.typography.displaySmall.copy(
+                      fontWeight = FontWeight.Bold, fontSize = 32.sp),
+              modifier = Modifier.testTag("quiz_title"))
+
+          // Earth Button
+          Button(
```

Comment:
It looks like both of these buttons for the "Earth" and "Solar System" quizzes have very similar implementations. You can refactor this code into a single composable function to avoid repetition and make the code easier to maintain in the future. Here’s a suggestion for refactoring:

```kotlin
@Composable
fun QuizOptionButton(text: String, onClick: () -> Unit, testTag: String) {
    Button(
        onClick = onClick,
        shape = RoundedCornerShape(16.dp),
        modifier = Modifier.fillMaxWidth().height(56.dp).testTag(testTag)
    ) {
        Text(text = text, fontSize = 20.sp, color = Color.White, fontWeight = FontWeight.SemiBold)
    }
}

// And then remove the two Button to just :  

QuizOptionButton(text = "Earth", onClick = { navigationActions.navigateTo(Screen.QUIZ_PLAY) }, "earth_button")
QuizOptionButton(text = "Solar System", onClick = { navigationActions.navigateTo(Screen.QUIZ_PLAY) }, "solar_system_button")

Reply:
Thank you for noticing this code duplication,  fixed it.